deploy/sre-prometheus/management-cluster/100-dynatrace-workloads-monitoring.PrometheusRule.yaml

```diff
@@ -0,0 +1,91 @@
+          kube_daemonset_status_desired_number_scheduled{daemonset=~".*-oneagent", namespace="dynatrace"} == 0 )
+      for: 15m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace Workloads are failing on the Management Cluster"
+        description: "The Dynatrace deployment on the Management Cluster has failed to be deployed for 15 mins"
+
+  - name: sre-dynatrace-monitoring-stack-degraded
+    rules:
+    - alert: DynatraceOperatorDegradedSRE
+      expr: (sum(kube_deployment_spec_replicas{deployment="dynatrace-operator", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_deployment_status_replicas_available{deployment="dynatrace-operator", namespace="dynatrace"}) without (deployment, instance, pod)) > 0
+      for: 10m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "The Dynatrace Operator is Degraded"
+        description: "{{$value}} pods are not running out of total pods for the Dynatrace Operator"
+  
+    - alert: DynatraceWebhookDegradedSRE
+      expr: (sum(kube_deployment_spec_replicas{deployment="dynatrace-webhook", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_deployment_status_replicas_available{deployment="dynatrace-webhook", namespace="dynatrace"}) without (deployment, instance, pod)) > 0
+      for: 10m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace Webhook is Degraded"
+        description: "{{$value}} pods are not running out of total pods for the Dynatrace Webhook"
+
+    - alert: DynatraceOpenTelemetryCollectorDegradedSRE
+      expr: (sum(kube_deployment_spec_replicas{deployment="opentelemetry-dynatrace-collector", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_deployment_status_replicas_available{deployment="opentelemetry-dynatrace-collector", namespace="dynatrace"}) without (deployment, instance, pod)) == 0
+      for: 15m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace OTEL Collector is Degraded"
+        description: "{{$value}} pods are not running out of total pods for the OpenTelemetry Dynatrace Collector"
+
+    - alert: DynatraceActiveGateDegradedSRE
+      expr: (sum(kube_statefulset_status_replicas{statefulset=~".*-activegate", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_statefulset_status_replicas_ready{statefulset=~".*-activegate", namespace="dynatrace"}) without (deployment, instance, pod)) > 0
+      for: 15m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace ActiveGate is Degraded"
+        description: "{{$value}} pods are not running out of total pods for Dynatrace ActiveGate"
+
+    - alert: DynatraceOneAgentDegraded
```

Comment:
Why does this alert not contain the SRE suffix while others do?

Reply:
also, should it be critical?

Reply:
Sure can add SRE suffix. For severity concern, the only thing I can think of is log based alerting. For OneAgent, an instance will run on a worker on MC cluster and through it the logs will be forwarded per node through ActiveGate to Dynatrace. However, if this single instance is not functional too, it would impact logs and a lot of node monitoring data that goes into Dynatrace. We can keep this as warning for now but can revisit the priority in future.